hera_pspec/pstokes.py

```diff
@@ -0,0 +1,139 @@
+"""
+Module to construct pseudo-Stokes (I,Q,U,V) visibilities from miriad files or UVData objects
+"""
+import numpy as np, os
+import pyuvdata
+import copy
+
+# weights used in forming Stokes visibilities
+pol_weights = {
+    'I': {'XX': 1. , 'YY': 1. },
+    'Q': {'XX': 1. , 'YY':-1. },
+    'U': {'XY': 1. , 'YX': 1. },
+    'V': {'XY':-1.j, 'YX': 1.j},
+}
+
+def miriad2pyuvdata(dset):
+   """
+   Converts Miriad file to UVData object
+
+   Parameters
+   ----------
+   dset : str
+      Miriad file to convert to UVData object containing visibilities and corresponding metadata
+   """
+   uv = pyuvdata.UVData()
+   uv.read_miriad(dset)
+   return uv
+
+def _combine_pol(uvd1, uvd2, pol1, pol2, pstokes='I'):
+   """
+   Reads in miriad file and combines visibilities to form the desired pseudo-stokes visibilities. It returns UVData object containing the pseudo-stokes visibilities
+   
+   Parameters
+   ---------
+   uvd1 : UVData object
+       First UVData object containing data that is used to
+       form Stokes visibilities
+
+   uvd2 : UVData oject
+       Second UVData objects containing data that is used to
+       form Stokes visibilities
+
+   pol1 : Polarization, type: str
+       Polarization of the first UVData object
+  
+   pol2 : Polarization, type: str
+       Polarization of the second UVData object
+
+   pstokes: Pseudo-stokes parameter, type: str
+       Pseudo stokes parameter to form, can be 'I' or 'Q' or 'U' or 'V'. Default: I
+  
+   """
+   if isinstance(uvd1, pyuvdata.UVData) == False:
+      raise TypeError("uvd1 must be a pyuvdata.UVData instance")
+   if isinstance(uvd2, pyuvdata.UVData) == False:
+      raise TypeError("uvd2 must be a pyuvdata.UVData instance")
+   
+   assert isinstance(pol1, str)
+   assert isinstance(pol2, str)
+
+   # extracting data array from the UVData objects
+   data1 = uvd1.data_array
+   data2 = uvd2.data_array
+
+   # extracting flag array from the UVdata objects
+   flag1 = uvd1.flag_array
+   flag2 = uvd2.flag_array
+   #constructing flags (boolean)
+   flag = flag1 + flag2
```

Comment:
Is this definitely always a Boolean operation? i.e. is `flag` still a boolean array after adding the two booleans?